### PR TITLE
kube-prometheus/docs: add docs on how to modify assets

### PR DESCRIPTION
**Pull request by brancz**
Closing #487

@Gouthamve @mxinden @fabxc 

/cc @gdmello

### CODE REVIEW

File: contrib/kube-prometheus/docs/developing-alerts-and-dashboards.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Developing Alerts and Dashboards
+
+`kube-prometheus` ships with a set of default alerting rules and dashboards. At some point one might like to extend them. This document is intended to explain the workflow of how additional alerting rules and dashboards could be added.
+
+For both, the Prometheus alerting rules as well as the Grafana dashboards, there are Kubernetes `ConfigMap`s, that are generated from content in the `assets/` directory.
+
+The source of truth for the alerts and dashboards are the files in the `assets/` directory. The respective files have to be changed there and then the `hack/scripts/generate-manifests.sh` script is executed to re-generate the Kubernetes manifests.
```
Comment (Gouthamve) [inline contrib/kube-prometheus/docs/developing-alerts-and-dashboards.md]: Can we have a makefile target for this? I think that is much more elegant.

Comment (brancz) [inline contrib/kube-prometheus/docs/developing-alerts-and-dashboards.md]: Done. Chose `make generate` to be consistent with the othe `Makefile`s.

**Comment by Gouthamve** [comment]
LGTM

**Comment by gdmello** [comment]
Thank you @brancz!

**Pull request merged**